### PR TITLE
migrate all setUp, helper functions and interfaces to a base contract

### PR DESCRIPTION
**Pull request by iamchrissmith**
# Description

Move spell setup and help functions to a base contract and inherit it. 

# Contribution Checklist

- [ ] PR title starts with `(PE-<TICKET_NUMBER>)`
- [ ] Code approved
- [ ] Tests approved
- [ ] CI Tests pass

# Checklist

- [ ] Every contract variable/method declared as public/external private/internal
- [ ] Consider if this PR needs the `officeHours` modifier
- [ ] Verify expiration (`4 days + 2 hours` monthly and `30 days` for the rest)
- [ ] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [ ] Validate all addresses used are in Goerli changelog or known
- [ ] Notify any external teams affected by the spell so they have the opportunity to review
- [ ] Deploy spell to Goerli `ETH_GAS="XXX" ETH_GAS_PRICE="YYY" make deploy`
- [ ] Ensure contract is verified on `Goerli` etherscan
- [ ] Change test to use Goerli spell address and deploy timestamp
- [ ] Keep `Goerli-DssSpell.sol` and `Goerli-DssSpell.t.sol` the same, but make a copy in `archive`
- [ ] `squash and merge` this PR

### CODE REVIEW

**Comment by naszam** [comment]
Test PASS (HardHat Fork pinned at block n. `5749312`, #22) 
```
dapp-test: rpc block: latest
Running 13 tests for src/Goerli-DssSpell.t.sol:DssSpellTest
[PASS] testFail_notScheduled() (gas: 12236)
[PASS] test_auth_in_sources() (gas: 3333266)
[PASS] test_use_eta() (gas: 337968)
[PASS] testOnTime() (gas: 476918)
[PASS] test_nextCastTime() (gas: 400437)
[PASS] testFailTooEarly() (gas: 372894)
[PASS] test_bytecode_matches() (gas: 905138)
[PASS] test_auth() (gas: 7074375)
[PASS] testSpellIsCast_GENERAL() (gas: 7482111)
[PASS] testFailWrongDay() (gas: 372947)
[PASS] testCastCost() (gas: 478716)
[PASS] testNewChainlogValues() (gas: 482610)
[PASS] testFailTooLate() (gas: 373237)
```

**Comment by iamchrissmith** [comment]
@gbalabasquer tagged you here since you are on spell duty this week.  I want to make sure i don't mess up your process, so figured it was better to wait until after your goerli work this week was done

File: src/Goerli-DssSpell.t.base.sol [hunk 1]
```diff
@@ -0,0 +1,2327 @@
...
+    function mint(uint256) external;
+    function mint(address,uint) external;
+    function burn(uint256) external;
+    function burn(address,uint) external;
+    function setName(bytes32) external;
+    function authority() external view returns (address);
+    function owner() external view returns (address);
+    function setOwner(address) external;
+    function setAuthority(address) external;
+}
```
Comment (naszam) [inline src/Goerli-DssSpell.t.base.sol:70]: We could move the interfaces in `dss-interfaces` repo to reduce space (`SpellLike`,` AuthLike` and `PsmAbstract`).
`BrokenTokenAbstract` I think could be removed as it was specific to #19 (USDT-A Offboarding).

Comment (iamchrissmith) [reply src/Goerli-DssSpell.t.base.sol:70]: i'll create another PR to handle that so this doesn't get too big

File: src/Goerli-DssSpell.t.base.sol [hunk 2]
```diff
@@ -0,0 +1,2327 @@
...
+                return;
+            } else {
+                // Keep going after restoring the original value
+                hevm.store(
+                    address(token),
+                    keccak256(abi.encode(address(this), uint256(i))),
+                    prevValue
+                );
+            }
+        }
```
Comment (naszam) [inline src/Goerli-DssSpell.t.base.sol:1941]: Same as above for `BrokenTokenAbstract`

**Pull request merged**